plugins/logistics.cpp

```diff
@@ -123,10 +129,13 @@ static void validate_stockpile_configs(color_ostream& out,
                 !c.get_bool(STOCKPILE_CONFIG_MELT) &&
                 !c.get_bool(STOCKPILE_CONFIG_TRADE) &&
                 !c.get_bool(STOCKPILE_CONFIG_DUMP) &&
-                !c.get_bool(STOCKPILE_CONFIG_TRAIN))) {
+                !c.get_bool(STOCKPILE_CONFIG_TRAIN) &&
+                !(c.get_int(STOCKPILE_CONFIG_FORBID) > 0))) {
             to_remove.push_back(stockpile_number);
             continue;
         }
+        if (c.get_int(STOCKPILE_CONFIG_FORBID) == -1)
+            c.set_int(STOCKPILE_CONFIG_FORBID, 0);
```

Comment:
This migration code should be in `plugin_load_site_data`, not here.

Reply:
Thank for this feedback, I was both overlooking and not understanding `plugin_load_site_data` 

I've moved the same logic to ~line 180 within `plugin_load_site_data` but I'm still not sure I understand,
```if (c.key() == CONFIG_KEY) continue;```

I put the mitigation after that and before,
```watched_stockpiles.emplace(c.get_int(STOCKPILE_CONFIG_STOCKPILE_NUMBER),  c);```

It seems to work there. But I presume that it is like `migrate_old_keys` in that the logic would be unnecessary if v50.current saves are no longer loadable? Should I include a comment to this effect?

Reply:
> `if (c.key() == CONFIG_KEY) continue;`

that line is there because I was silly with the naming and the plugin-wide config is also matched by CONFIG_KEY_PREFIX. We don't want to treat the plugin-wide config as a stockpile-specific config. They're different schemas.